init/setup-rwdev.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+set -e
+
+if [ -e /dev/xvdb ] ; then
+    # The private /dev/xvdb device is present.
+
+    # check if private.img (xvdb) is empty - all zeros
+    private_size_512=`blockdev --getsz /dev/xvdb`
+    if dd if=/dev/zero bs=512 count=$private_size_512 2>/dev/null | diff /dev/xvdb - >/dev/null; then
+        # the device is empty, create filesystem
+        echo "Virgin boot of the VM: creating private.img filesystem" >&2
+        mkfs.ext4 -m 0 -q /dev/xvdb || exit 1
+    fi
+
+    tune2fs -m 0 /dev/xvdb
+    echo "Virgin boot of the VM: marking private.img as clean" >&2
+    fsck.ext4 -fp /dev/xvdb
```

Comment:
Why `fsck.ext4 -f` on every boot? If it was cleanly unmounted, it shouldn't be needed, it will only slow down the startup. If required by `resize2fs`, it should be called only when really resize is needed.
So, maybe drop `-f`? Or run with `-f` only if resize is needed?